document/paragraphproperties.go

```diff
@@ -199,3 +199,186 @@ func (p ParagraphProperties) SetHangingIndent(m measurement.Distance) {
+func (p ParagraphProperties) EastAsiaFont() string {
+	if fonts := p.x.RPr.RFonts; fonts != nil {
+		if fonts.EastAsiaAttr != nil {
+			return *fonts.EastAsiaAttr
+		}
+	}
+	return ""
+}
+
+// ColorStr returns the hex value of paragraph color.
+func (p ParagraphProperties) ColorStr() string {
+	if color := p.x.RPr.Color; color != nil {
+		valAttr := color.ValAttr
+		if valAttr.ST_HexColorRGB != nil {
+			return *valAttr.ST_HexColorRGB
+		}
+	}
+	return ""
+}
+
+// SpacingValue returns the value of paragraph spacing.
+func (p ParagraphProperties) SpacingValue() int64 {
+	if spacing := p.x.RPr.Spacing; spacing != nil {
+		valAttr := spacing.ValAttr
+		if valAttr.Int64 != nil {
+			return *valAttr.Int64
+		}
+	}
+	return int64(0)
+}
+
+// SpacingMeasure returns the measure of paragraph spacing.
+func (p ParagraphProperties) SpacingMeasure() string {
+	if spacing := p.x.RPr.Spacing; spacing != nil {
+		valAttr := spacing.ValAttr
+		if valAttr.ST_UniversalMeasure != nil {
+			return *valAttr.ST_UniversalMeasure
+		}
+	}
+	return ""
+}
+
+// SizeValue returns the value of paragraph font size.
+func (p ParagraphProperties) SizeValue() float64 {
+	if sz := p.x.RPr.Sz; sz != nil {
+		valAttr := sz.ValAttr
+		if valAttr.ST_UnsignedDecimalNumber != nil {
+			return float64(*valAttr.ST_UnsignedDecimalNumber) / 2
+		}
+	}
+	return 0.0
+}
+
+// SizeMeasure returns the measure of paragraph font size.
+func (p ParagraphProperties) SizeMeasure() string {
+	if sz := p.x.RPr.Sz; sz != nil {
+		valAttr := sz.ValAttr
+		if valAttr.ST_PositiveUniversalMeasure != nil {
+			return *valAttr.ST_PositiveUniversalMeasure
+		}
+	}
+	return ""
+}
+
+// ComplexSizeValue returns the value of paragraph font size for complex fonts.
+func (p ParagraphProperties) ComplexSizeValue() float64 {
+	if szCs := p.x.RPr.SzCs; szCs != nil {
+		valAttr := szCs.ValAttr
+		if valAttr.ST_UnsignedDecimalNumber != nil {
+			return float64(*valAttr.ST_UnsignedDecimalNumber) / 2
+		}
+	}
+	return 0.0
+}
+
+// ComplexSizeMeasure returns the measure of paragraph font size for complex fonts.
+func (p ParagraphProperties) ComplexSizeMeasure() string {
+	if szCs := p.x.RPr.SzCs; szCs != nil {
+		valAttr := szCs.ValAttr
+		if valAttr.ST_PositiveUniversalMeasure != nil {
+			return *valAttr.ST_PositiveUniversalMeasure
+		}
+	}
+	return ""
+}
+
+// UnderlineType returns the type of paragraph underline.
+func (p ParagraphProperties) UnderlineType() wml.ST_Underline {
+	if underline := p.x.RPr.U; underline != nil {
+		return underline.ValAttr
+	}
+	return 0
+}
+
+// UnderlineColor returns the hex color value of paragraph underline.
+func (p ParagraphProperties) UnderlineColor() string {
+	if underline := p.x.RPr.U; underline != nil {
+		color := underline.ColorAttr
+		if color.ST_HexColorRGB != nil {
+			return *color.ST_HexColorRGB
+		}
+	}
+	return ""
+}
+
+// VertAlign returns the value of paragraph vertical align.
+func (p ParagraphProperties) VertAlign() sharedTypes.ST_VerticalAlignRun {
```

Comment:
Same consideration here regarding encapsulation by higher level type

Reply:
Related in codebase:
```go
func (r RunProperties) SetVerticalAlignment(v sharedTypes.ST_VerticalAlignRun) {
```
So for consistency this should be called
```VerticalAlignment()```
Probably no need to include in a higher level type